From 53642bea9d41cba9531764e7031ded08b3324ade Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Tue, 12 Jan 2016 14:51:05 +0200
Subject: [PATCH 0669/1240] fix: io: Fix register access macros when compiled
 with REG_DEBUG

	The return value of the macro should be last as last operation,
	before the printout macro call.

Change-Id: Ia8e0be154b99eeac045f9f9bf65f1c0f983fa254
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/26734
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
---
 arch/arm/include/asm/io.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/include/asm/io.h b/arch/arm/include/asm/io.h
index dc623e8..c977348 100644
--- a/arch/arm/include/asm/io.h
+++ b/arch/arm/include/asm/io.h
@@ -157,10 +157,10 @@ static inline void __raw_readsl(unsigned long addr, void *data, int longlen)
 #define __iormb()	dmb()
 #define __iowmb()	dmb()
 
-#define writeb(v,c)	({ u8  __v = v; __iowmb(); __arch_putb(__v,c); __v; dbg_wb(__v,c);})
-#define writew(v,c)	({ u16 __v = v; __iowmb(); __arch_putw(__v,c); __v; dbg_ww(__v,c);})
-#define writel(v,c)	({ u32 __v = v; __iowmb(); __arch_putl(__v,c); __v; dbg_wl(__v,c);})
-#define writeq(v,c)	({ u64 __v = v; __iowmb(); __arch_putq(__v,c); __v; })
+#define writeb(v, c)	({ u8  __v = v; __iowmb(); __arch_putb(__v, c); dbg_wb(__v, c); __v; })
+#define writew(v, c)	({ u16 __v = v; __iowmb(); __arch_putw(__v, c); dbg_ww(__v, c); __v; })
+#define writel(v, c)	({ u32 __v = v; __iowmb(); __arch_putl(__v, c); dbg_wl(__v, c); __v; })
+#define writeq(v, c)	({ u64 __v = v; __iowmb(); __arch_putq(__v, c); __v; })
 
 #define readb(c)	({ u8  __v = __arch_getb(c); __iormb();  dbg_rb(__v,c); __v; })
 #define readw(c)	({ u16 __v = __arch_getw(c); __iormb();  dbg_rw(__v,c); __v; })
-- 
1.9.1

